### PR TITLE
Update Rust to 1.66 and fix some lints

### PR DESCRIPTION
**Pull request by utam0k**

### CODE REVIEW

**Comment by dcantah** [comment]
Clippy angry about one more unnecessary cast: https://github.com/containerd/rust-extensions/blob/b58e93208e1b1c85defb09f6abff177f529fa88d/crates/runc-shim/src/synchronous/task.rs#L363

**Comment by utam0k** [comment]
:wave: @dcantah Thanks for your review. I have fixed it.

**Comment by dcantah** [comment]
Thanks, I'll try and take a look sometime today

File: crates/client/vendor/github.com/containerd/containerd/api/services/events/v1/events.proto [hunk 1]
```diff
@@ -45,7 +45,7 @@ service Events {
 	//
 	// Unlike many other methods in containerd, subscribers will get messages
 	// from all namespaces unless otherwise specified. If this is not desired,
-	// a filter can be provided in the format 'namespace==<namespace>' to
+	// a filter can be provided in the format 'namespace=={namespace}' to
```
Comment (dcantah) [inline crates/client/vendor/github.com/containerd/containerd/api/services/events/v1/events.proto:48]: Should we change this upstream first (or just in parallel)? README states this is just a copy paste and not some tooling doing this, but better to keep these in sync even if it's just a comment touch up.

**Review by dcantah — APPROVED** [review]
Seems the CI failures are known (and you reported them :) https://github.com/containerd/rust-extensions/issues/110) and the changes in this PR didn't cause them.

**Review by Burning1020 — APPROVED** [review]
LGTM

**Comment by Burning1020** [comment]
I believe these two are not caused by you, but I can't figure out why it failed either.

File: rust-toolchain.toml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.65"
```
Comment (mxpv) [inline rust-toolchain.toml:2]: `clippy.toml` also needs update (also it might trigger some new lints)

Comment (utam0k) [reply rust-toolchain.toml:2]: fixed!

**Review by mxpv — APPROVED** [review]

**Pull request merged**